sklearn/tree/_export.py

```diff
@@ -75,6 +77,23 @@ def __repr__(self):
 SENTINEL = Sentinel()
 
 
+@validate_params(
+    {
+        "decision_tree": [DecisionTreeClassifier, DecisionTreeRegressor],
+        "max_depth": [Interval(Integral, 0, None, closed="left"), None],
+        "feature_names": [list, None],
+        "class_names": [list, None],
+        "label": [StrOptions({"all", "root", "none"})],
+        "filled": ["boolean"],
+        "impurity": ["boolean"],
+        "node_ids": ["boolean"],
+        "proportion": ["boolean"],
+        "rounded": ["boolean"],
+        "precision": [Interval(Integral, 0, None, closed="left"), None],
+        "ax": [plt.Figure, None],
```

Comment:
We don't want to import matplotlib by default because it's not a hard dependency of scikit-learn. This is why we import it only in the export method. I'd rather delegate the validation to matplotlib for this param

```suggestion
        "ax": "no_validation",  # delegate validation to matplotlib
```

What do you think @glemaitre ?

Reply:
ok.
"How would Typing help here ?" >>
just as a way to inform the user of the expected type.

Reply:
Anyway this is not super crucial so we can decide to merge it as it is, without the Figure validation for now.
@jeremiedbb

Reply:
I still see "Merging is blocked
The base branch restricts merging to authorized users. [Learn more about protected branches.](https://docs.github.com/articles/about-protected-branches/)"
Even though the pr seems to be approved.

Reply:
Only maintainers can merge PRs. This is the reason you see this message.